src/web_app/js/peerconnectionclient.js

```diff
@@ -358,7 +361,76 @@ PeerConnectionClient.prototype.onRemoteStreamAdded_ = function(event) {
 };
 
 PeerConnectionClient.prototype.onError_ = function(tag, error) {
+  var localSdp = this.localSdp || null;
+  var remoteSdp = this.remoteSdp || null;
   if (this.onerror) {
     this.onerror(tag + ': ' + error.toString());
+    this.callStats_.reportError(this.pc_, this.conferenceId_, tag, error,
+        localSdp, remoteSdp);
   }
 };
+
+// Setup the callstats api and attach it to the peerconnection.
+PeerConnectionClient.prototype.setupCallStats_ = function() {
+  if (typeof $ !== 'function' && typeof io !== 'function' &&
+      typeof jsSHA !== 'function')  {
+    trace('Callstats dependencies missing, stats will not be setup.');
+    return;
+  }
+
+  trace('Setting up callstats.');
+  // jscs:disable requireCapitalizedConstructors
+  /* jshint newcap: false */
+  var callStats = new callstats($, io, jsSHA);
+  // jscs:enable requireCapitalizedConstructors
+  /* jshint newcap: true */
+
+  var appId = '423310139';
+  var appSecret = 'jpZu5pquoROlCn5gekQX1RzTHPk=';
+  var userId = this.params_.clientId;
```

Comment:
I've set the originURL in the callstats dashboard setting ofc. Regarding the secrets, does it matter as long as we have the originURL restriction in place?